src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/mappers/McCostParamsMapper.java

```diff
@@ -25,8 +31,16 @@ public static McCostParams map(RoutingRequest request) {
 
     builder.wheelchairAccessibility(request.wheelchairAccessibility);
 
-    builder.unpreferredRoutes(request.unpreferredRoutes.stream().collect(Collectors.toSet()));
-    builder.unpreferredCost(request.unpreferredRouteCost);
+    var unpreferredRoutes = request
+      .getUnpreferredAgencies()
+      .stream()
+      .map(routesByAgency)
+      .flatMap(Collection::stream)
+      .collect(Collectors.toSet());
```

Comment:
Would it make sense to do this already in RoutingRequest?

Reply:
Wouldn't that mean that the RoutingRequest has a dependency in the TransitModelIndex?

Reply:
The RoutingRequest should be a POJO - as little logic as possible, and no dependencies on transit model/data.

Reply:
Aren't you saying that is should _not_ move to the RoutingRequest?

Reply:
I think this belong here, not in the RoutingRequest, the RoutingRequest should be "user-friendly" and not tied to implementation - mapping into the domain(raptor search) happens in this mapper.